tests/behat/features/capabilities/groups/join/groups-join-direct.feature

```diff
@@ -0,0 +1,12 @@
+@api
+Feature: A group can be configured to allow joining directly
+
+  Scenario: As anonymous user view a public group
+    Given groups with non-anonymous owner:
+      | label      | field_group_description | type           | langcode | field_flexible_group_visibility | field_group_allowed_join_method |
+      | Test group | Public visibility       | flexible_group | en       | public                          | direct                          |
+    And I am an anonymous user
+
+    When I am viewing the "stream" page of group "Test group"
+
+    Then I should see "Test group"
```

Comment:
As this file is under join folder, here I believe you could also check the join method by checking if the text or button with the name "Join" is present on the page